pkg/model/components/context.go

```diff
@@ -176,13 +169,13 @@ func Image(component string, architecture string, clusterSpec *kops.ClusterSpec,
 	//
 	// But ... this is only the case from 1.16 on...
 	if kubernetesVersion.IsGTE("1.16") {
-		imageName += "-" + architecture
+		imageName += "-amd64"
```

Comment:
Why hardcode amd64 here and below?

Reply:
This is already hardcoded to `amd64`:
https://github.com/kubernetes/kops/blob/bfd65ae589af979f0df9252b5269dcb5b3eb700b/pkg/model/components/context.go#L57-L59

Even more, this part of the code is used only for side-loading during CI, which works only with `amd64` images.

The approach taken for multi-arch support is to provide the list of assets for all supported architectures and let NodeUp decide which ones to download. At this point, it wouldn't really be possible to know which architecture would be used.

Reply:
It's not immediately obvious to me why sideloading would only work with amd64. I presume making `LoadImageTask` not sideload the amd64 images on arm would be a later PR, but I don't see it in the WIP PR.

I would have thought we'd want to support arm-only CI builds, in which case we'd get `OptionsContext.Architecture()` to return the target architecture.

Reply:
At the moment there are not CI binaries for arm64.
AMD64:
https://storage.googleapis.com/kubernetes-release-dev/ci/v1.19.0-beta.0.313+46d08c89ab9f55/bin/linux/amd64/kube-apiserver.tar
ARM64 (doesn't exists):
https://storage.googleapis.com/kubernetes-release-dev/ci/v1.19.0-beta.0.313+46d08c89ab9f55/bin/linux/arm64/kube-apiserver.tar

Reply:
It will work though, by using `imageName += "-{{arch}}"` and later replacing it in NodeUp with the detected architecture.

Reply:
But the different arch images will have different hashes.

Reply:
It looks like this isn't for sideloading but for filling in the various `Image` fields in the full cluster spec. That entire concept is going to have to be adjusted, at least in the case of kube-proxy, for multi-arch.